src/main/java/org/commcare/formplayer/services/CaseSearchHelper.java

```diff
@@ -209,9 +213,14 @@ private String getCacheKey(String url, Multimap<String, String> queryParams) thr
             builder.append("_").append(restoreFactory.getAsUsername());
         }
         builder.append("_").append(uri);
-        for (String key : queryParams.keySet()) {
+        List<String> keyList = queryParams.keySet().stream().collect(Collectors.toList());
+        Collections.sort(keyList);
+        for (String key : keyList) {
             builder.append("_").append(key);
-            for (String value : queryParams.get(key)) {
+            Collection<String> values = queryParams.get(key);
+            List<String> valuesList = values.stream().collect(Collectors.toList());
+            Collections.sort(valuesList);
```

Comment:
You can combine the sorting with the previous operation:
```suggestion
            List<String> valuesList = values.stream().sorted().toList();
```